### PR TITLE
Fix pretty printing of quotes in `mocha_inspect`

### PR DESCRIPTION
**Pull request by neonichu**
The pretty-printing of quotes from `inspect` was messing up actual quotes a bit, consider this example:

```ruby
#!/usr/bin/env ruby

require 'bacon'
require 'mocha-on-bacon'
require 'open4'

Open4.expects(:spawn).with('"').once
Open4.spawn('\'')
```

before this PR:

```
./quotes.rb:8:in `<main>': unexpected invocation: Open4.spawn(''') (Mocha::ExpectationError)
unsatisfied expectations:
- expected exactly once, not yet invoked: Open4.spawn('\'')
```

after this PR:

```
./quotes.rb:8:in `<main>': unexpected invocation: Open4.spawn('\'') (Mocha::ExpectationError)
unsatisfied expectations:
- expected exactly once, not yet invoked: Open4.spawn('"')
```

So one now no longer gets confused by replaced double-quotes or incorrectly escaped single-quotes in the messages.

### CODE REVIEW

**Comment by floehopper** [comment]
@neonichu: Thanks for taking the time to submit this - it does look as if you've identified a genuine problem. I'll try to have a proper look as soon as I can. At first glance, the new implementation is hard to follow. Also it's not obvious that the two new unit tests are sufficient to require all of the logic in the new implementation. Writing more exhaustive unit tests might help explain the implementation better.

**Comment by floehopper** [comment]
@neonichu: I'm so sorry not to have looked at this for so long. I've had a bit of a think and come up with a possible alternative solution in #223. Can you see whether that would address your issue? Thanks.

**Comment by neonichu** [comment]
Superseded by #223
